includes/admin/upgrades/v3/class-data-migrator.php

```diff
@@ -479,13 +479,13 @@ public static function orders( $data = null ) {
 			$order_tax = (float) $meta_tax[0];
 		}
 
+		$meta_total = false;
 		// Retrieve the total amount from metadata if available.
-		$meta_total = isset( $meta['_edd_payment_total'] )
-			? $meta['_edd_payment_total']
-			: false;
-
-		if ( false !== $meta_total ) {
-			$meta_total  = maybe_unserialize( $meta_total );
+		if ( isset( $meta['_edd_payment_total'] ) ) {
+			$meta_total  = maybe_unserialize( $meta['_edd_payment_total'] );
+			$order_total = (float) $meta_total[0];
+		} elseif ( isset( $payment_meta['amount'] ) ) {
+			$meta_total  = maybe_unserialize( $payment_meta['amount'] );
 			$order_total = (float) $meta_total[0];
```

Comment:
```suggestion
			$order_total = (float) $meta_total;
```

Data retrieved from `$meta` is an array, but `$payment_meta` is not.

Reply:
Patched that change in locally and it's now working well for me. So I'll be happy to merge this after that change.

Reply:
Confirmed here--my $7.50 order was showing in 3.0 as $7, but doing this fixed it.